app/client/keybase/keystore.go

```diff
@@ -0,0 +1,391 @@
+func (keybase *badgerKeybase) GetPrivKey(address, passphrase string) (crypto.PrivateKey, error) {
+	kp, err := keybase.Get(address)
+	if err != nil {
+		return nil, err
+	}
+
+	privKey, err := kp.Unarmour(passphrase)
+	if err != nil {
+		return nil, err
+	}
+
+	return privKey, nil
+}
+
+// Get all the addresses and key pairs stored in the keybase
+// Returns addresses stored and all the KeyPair structs stored in the DB
+func (keybase *badgerKeybase) GetAll() (addresses []string, keyPairs []KeyPair, err error) {
+	// View executes the function provided managing a read only transaction
+	err = keybase.db.View(func(tx *badger.Txn) error {
+		opts := badger.DefaultIteratorOptions
+		opts.PrefetchSize = 5
+		it := tx.NewIterator(opts)
+		defer it.Close()
+		for it.Rewind(); it.Valid(); it.Next() {
+			item := it.Item()
+			err := item.Value(func(val []byte) error {
+				b := make([]byte, len(val))
+				copy(b, val)
+				// Decode []byte value back into KeyPair struct
+				var kp KeyPair
+				bz := new(bytes.Buffer)
+				bz.Write(b)
+				dec := gob.NewDecoder(bz)
+				if err = dec.Decode(&kp); err != nil {
+					return err
+				}
+
+				addresses = append(addresses, kp.GetAddressString())
+				keyPairs = append(keyPairs, kp)
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return addresses, keyPairs, nil
+}
+
+// Check whether an address is currently stored in the DB
+func (keybase *badgerKeybase) Exists(address string) (bool, error) {
+	val, err := keybase.Get(address)
+	if err != nil {
+		return false, err
+	}
+	return val != KeyPair{}, nil
+}
+
+// Export the Private Key string of the given address
+func (keybase *badgerKeybase) ExportPrivString(address, passphrase string) (string, error) {
+	kp, err := keybase.Get(address)
+	if err != nil {
+		return "", err
+	}
+	return kp.ExportString(passphrase)
+}
+
+// Export the Private Key of the given address as a JSON object
+func (keybase *badgerKeybase) ExportPrivJSON(address, passphrase string) (string, error) {
+	kp, err := keybase.Get(address)
+	if err != nil {
+		return "", err
+	}
+	return kp.ExportJSON(passphrase)
+}
+
+func (keybase *badgerKeybase) UpdatePassphrase(address, oldPassphrase, newPassphrase string) error {
+	// Check the oldPassphrase is correct
+	privKey, err := keybase.GetPrivKey(address, oldPassphrase)
+	if err != nil {
+		return err
+	}
+	privStr := privKey.String()
+
+	addrBz, err := hex.DecodeString(address)
+	if err != nil {
+		return err
+	}
+
+	err = keybase.db.Update(func(tx *badger.Txn) error {
+		keyPair, err := CreateNewKeyFromString(privStr, newPassphrase)
+		if err != nil {
+			return err
+		}
+
+		// Use key address as key in DB
+		key := keyPair.GetAddressBytes()
+		if bytes.Compare(key, addrBz) != 0 {
+			return fmt.Errorf("Key address does not match previous address.")
+		}
+		// Encode entire KeyPair struct into []byte for value
+		bz := new(bytes.Buffer)
```

Comment:
s/bz/keypairBz

These small nits go a long way when it comes to spending less time of other devs.

Just leaving one comment to avoid being pedantic, but use your best judgment elsewhere